auth/auth.ts

```diff
@@ -169,46 +176,12 @@ export class MockAuth implements firebase.auth.Auth {
     throw new Error("Method not implemented.");
   }
 
-  async signInWithPopup(
-    provider: firebase.auth.AuthProvider
-  ): Promise<firebase.auth.UserCredential> {
-    const mock = Array.from(this.socialSignIns.values()).find(
-      (mock) => mock.type === provider.providerId
-    );
-
-    if (!mock) {
-      throw new Error("No mock response set.");
-    }
-
-    // Mock is used, then it must go
-    this.socialSignIns.delete(mock);
-
-    const data = await mock.response;
-    let user = this.store.findByEmail(data.email);
-    if (user) {
-      if (user.providerId !== provider.providerId) {
-        throw new Error("auth/account-exists-with-different-credential");
-      }
-
-      return this.signIn(user, {
-        isNewUser: false,
-        providerId: provider.providerId,
-        profile: null,
-        username: data.email,
-      });
-    }
-
-    user = this.store.add({ ...data, providerId: provider.providerId });
-    return this.signIn(user, {
-      isNewUser: true,
-      providerId: provider.providerId,
-      profile: null,
-      username: data.email,
-    });
+  signInWithPopup(provider: firebase.auth.AuthProvider): Promise<firebase.auth.UserCredential> {
+    return this.signInWithSocial(provider);
   }
 
-  signInWithRedirect(provider: firebase.auth.AuthProvider): Promise<void> {
-    throw new Error("Method not implemented.");
+  async signInWithRedirect(provider: firebase.auth.AuthProvider): Promise<void> {
+    await this.signInWithSocial(provider);
```

Comment:
Can we consider this a bonus new feature?

Reply:
😅  not sure what you mean by bonus, however I believe this should set a value that `getRedirectResult()` can retrieve to mimic the behavior of Firebase.

```
signInWithRedirect ( provider :  AuthProvider ) : Promise < void >
Authenticates a Firebase client using a full-page redirect flow. To handle the results and errors for this operation, refer to firebase.auth.Auth.getRedirectResult.
```

Reply:
Ah cool. Didn't know about that method!
Looks like it's a bit more evolved than just doing the underlying social sign in process. Probably worth extracting into its own PR.

Reply:
Reverted.